### PR TITLE
add support for 'only_if_executable' and handle predefined OSS configs differently

### PR DESCRIPTION
**Pull request by ThePrez**

### CODE REVIEW

**Comment by chrjorgensen** [comment]
Hi Jesse.

Seems like this PR somehow obsoletes my later PR #140 ... :grin:  ...and also my later PR #141 ...! :joy: 

But not completely, since I also changed the PostgreSQL configuration in #140.
So the PR's last merged must be updated before merge to reflect the changes in the first merged PR...!

I had a quick look, and as I understand your change, you want to always deploy the OSS configurations previously found in the samples directory and let the Service Commander check for the executable before start and stop of the service. Right?

One question then: How do you manage any user changes to the configs? That they are not overwritten in future updates of Service Commander? Or do the users have to copy the config files to another directory to keep their changes - a directory that Service Commander will read before its own "oss-common" directory?

File: service-commander.spec [hunk 1]
```diff
@@ -99,7 +100,7 @@ fi
 - feature: Several new options for 'sc_install_defaults'
 - feature: New '--ignore-groups' option for 'sc'
 - feature: Allow service path names to be relative
-- feature: preinstall common system services
+- feature: preinstall crond, MariaDB, and Zend DBi
```
Comment (chrjorgensen) [inline service-commander.spec:103]: How about PostgreSQL - is it too difficult to include in the OSS common services? Or maybe we can do that when we have resolved the PR "dispute"... :grin: 

By the way, there is one comma too many...

File: samples/oss_common/mariadb.yaml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 name: MariaDB Server
 start_cmd: /QOpenSys/pkgs/bin/mysqld_safe --datadir=/QOpenSys/var/lib/mariadb/data
-
+only_if_executable: /QOpenSys/pkgs/bin/mysqld_safe
 check_alive: port
 check_alive_criteria: 3306
```
Comment (chrjorgensen) [inline samples/oss_common/mariadb.yaml:5]: check_alive_criteria is deprecated and should be avoided, right? I fixed this in PR #141.

File: samples/postgres.yaml [hunk 3]
```diff
@@ -1,6 +1,7 @@
 name: PostgreSQL Server
-start_cmd: pg_ctl start
-stop_cmd: pg_ctl stop
+start_cmd: /QOpenSys/pkgs/bin/pg_ctl start
+stop_cmd: /QOpenSys/pkgs/bin/pg_ctl stop
+only_if_executable: /QOpenSys/pkgs/bin/pg_ctl
 
 check_alive: port
```
Comment (chrjorgensen) [inline samples/postgres.yaml:6]: I changed the PostgreSQL configuration in PR #140...

**Review by chrjorgensen — COMMENTED** [review]
Hi Jesse.

I ran through your changes and made some comments.

I didn't check the Java source files and I haven't tested your changes on my little IBM i "sandbox" - I will do that when we have discussed my comments. OK? :question:

File: service-commander.spec (hunk 1 above)
Comment (ThePrez) [reply service-commander.spec:103]: Let's continue that discussion in #142

**Pull request closed without merging**